lib/bolt/shell/bash.rb

```diff
@@ -53,19 +53,35 @@ def upload(source, destination, options = {})
 
       def download(source, destination, options = {})
         running_as(options[:run_as]) do
-          # Target OS may be either Unix or Windows. Without knowing the target OS before-hand
-          # we can't assume whether the path separator is '/' or '\'. Assume we're connecting
-          # to a target with Unix and then check if the path exists after downloading.
           download = File.join(destination, Bolt::Util.unix_basename(source))
 
-          conn.download_file(source, destination, download)
+          # If using run-as, the file is copied to a tmpdir and chowned to the
+          # connecting user. This is a workaround for limitations in net-ssh that
```

Comment:
I'm honestly not sure that `net-ssh` should support a run-as user, this use case is somewhat specific to Bolt.